### PR TITLE
Initial commit of admin panel

### PR DESCRIPTION
**Pull request by caseyswilliams**
- Created admin section
- Created initial user admin page, only has ability to set flags at the moment
- Most of the changes are just changes to imports to use @ alias because I hate relative paths, I should've done this sooner

### CODE REVIEW

**Comment by Saibamen** [comment]
Related: #135

File: UT4MasterServer.Web/src/components/Header.vue [hunk 1]
```diff
@@ -53,13 +61,12 @@
   justify-content: flex-end;
 
   .nav-link.router-link-active,
-  .navbar-nav .show > .nav-link {
+  .navbar-nav .show>.nav-link {
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/components/Header.vue:64]: Why?

File: UT4MasterServer.Web/src/enums/account-flag.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export enum AccountFlag {
+    None = 'None',
+    Admin = 'Admin',
+    Moderator = 'Moderator',
+    Developer = 'Developer',
+    ContentCreator = 'ContentCreator',
+    HubOwner = 'HubOwner'
+}
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/enums/account-flag.ts:8]: Add numbers in comments from `UT4MasterServer.Models.AccountFlags` enum

File: UT4MasterServer.Web/src/route-guards.ts [hunk 3]
```diff
@@ -19,3 +19,10 @@ export async function privateGuard(to: RouteLocationNormalized) {
     return { path: '/Login' };
   }
 }
+
+export async function adminGuard(to: RouteLocationNormalized) {
+  if (AccountStore.account === null) {
+    await  AccountStore.fetchUserAccount();
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/route-guards.ts:25]:
```suggestion
    await AccountStore.fetchUserAccount();
```

File: UT4MasterServer.Web/src/services/http.service.ts [hunk 4]
```diff
@@ -19,13 +19,14 @@ export default class HttpService {
...
+    async send<K = unknown, T = unknown>(url: string, options?: HttpRequestOptions<T>, method: HttpMethod = 'GET', form = true): Promise<K> {
         const fetchOptions: RequestInit = { method };
         if (options?.body) {
-            fetchOptions.body = this.formEncode(options.body);
+            fetchOptions.body = form ? this.formEncode(options.body) : JSON.stringify(options.body);
         }
 
-        const headers: HeadersInit = { 'Content-Type': 'application/x-www-form-urlencoded' };
+        const headers: HeadersInit = { 'Content-Type': form ? 'application/x-www-form-urlencoded' : 'application/json' };
+
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/services/http.service.ts:29]: Double line
```suggestion
        const headers: HeadersInit = { 'Content-Type': form ? 'application/x-www-form-urlencoded' : 'application/json' };
```

File: UT4MasterServer.Web/src/components/Header.vue (hunk 1 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/components/Header.vue:64]: Must've autoformatted :shrug:

File: UT4MasterServer.Web/src/enums/account-flag.ts (hunk 2 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/enums/account-flag.ts:8]: The enum numbers aren't relevant to the front end

File: UT4MasterServer.Web/src/services/http.service.ts (hunk 4 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/services/http.service.ts:29]: I just autoformat with vscode, it wraps when it deems the line too long.

Comment (caseyswilliams) [reply UT4MasterServer.Web/src/services/http.service.ts:29]: I'll add prettier in a separate PR, that should alleviate all of your whitespace concerns.

File: UT4MasterServer/Controllers/AccountController.cs [hunk 5]
```diff
@@ -97,6 +97,21 @@ public async Task<IActionResult> GetAccount(string id)
...
+		for (int i = 0; i < flagNamesAll.Length; i++)
+		{
+			if (account.Flags.HasFlag(flagValuesAll[i]))
+			{
+				flags.Add(flagNamesAll[i]);
+			}
+		}
+		var flagsArray = new JArray(flags);
+		obj.Add("flags", flagsArray);
+
```
Comment (timiimit) [inline UT4MasterServer/Controllers/AccountController.cs:114]: This endpoint is used by the game and should not be changed. From the looks of it, it seems that what you added here is never even needed anywhere.

File: UT4MasterServer.Web/src/types/account.ts [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
+    preferredLanguage?: string;
+    canUpdateDisplayName?: boolean;
+    tfaEnabled?: boolean;
+    emailVerified?: boolean;
+    minorVerified?: boolean;
+    minorExpected?: boolean;
+    minorStatus?: string;
+    cabinedMode?: boolean;
+    hasHashedEmail?: boolean
+    flags?: AccountFlag[];
```
Comment (timiimit) [inline UT4MasterServer.Web/src/types/account.ts:24]: `flags` should not be returned in the standard account fetching endpoint, so i assume this shouldn't be here as well.

File: UT4MasterServer/Controllers/AccountController.cs (hunk 5 above)
Comment (caseyswilliams) [reply UT4MasterServer/Controllers/AccountController.cs:114]: I use it in the front end, but I can put it in another place. Persona controller?

File: UT4MasterServer.Web/src/types/account.ts (hunk 6 above)
Comment (caseyswilliams) [reply UT4MasterServer.Web/src/types/account.ts:24]: Yeah I'm gonna have to move the whole thing to a different controller action, that's fine. I'd rather avoid using any of the epic endpoints so I don't have to worry about changing them

Comment (caseyswilliams) [reply UT4MasterServer.Web/src/types/account.ts:24]: We should maybe mark the ones used by the game so we know not to change em

File: UT4MasterServer.Web/src/pages/Profile/PlayerCard.vue [hunk 7]
```diff
@@ -66,21 +66,5 @@ img.avatar {
...
-async function fetchProfile() {
-  if (SessionStore.session) {
-    playerCard.value = await customService.getPlayerCard(SessionStore.session.account_id);
-  }
-}
-
-onMounted(fetchProfile);
-
+import { AccountStore } from '@/stores/account-store';
 </script>
```
Comment (Saibamen) [inline UT4MasterServer.Web/src/pages/Profile/PlayerCard.vue:70]: We will use `playerCard` to also show other players cards in game.
IMO we should have deleted `playerCard` constant here, to be able to modify it when we get player ID from GET (original Epic URL: https://www.epicgames.com/unrealtournament/en-US/playerCard?playerId=63e4964c6cc1441089a726c00fcb2926)

**Review by timiimit — APPROVED** [review]

File: UT4MasterServer.Web/src/services/http.service.ts (hunk 4 above)
Comment (Saibamen) [reply UT4MasterServer.Web/src/services/http.service.ts:29]: > I just autoformat with vscode, it wraps when it deems the line too long.

I selected 2 lines and deleted blank one in suggestion, but GH don't get it...
You have double lines - 29 and 30

**Pull request merged**